src/reaction_lammps_mupt/input_parser.py

```
@@ -0,0 +1,103 @@
+
+        if not isinstance(inputs["monomers"], dict) or len(inputs["monomers"]) == 0:
+            raise ValueError("The 'monomers' key must map to a non-empty dictionary.")
+
+        if not isinstance(inputs["number_of_monomers"], dict) or len(inputs["number_of_monomers"]) == 0:
+            raise ValueError("The 'number_of_monomers' key must map to a non-empty dictionary.")
+
+        if len(inputs["monomers"]) != len(inputs["number_of_monomers"]):
+            raise ValueError(
+                f"Length mismatch: 'monomers' has {len(inputs['monomers'])} entries "
+                f"but 'number_of_monomers' has {len(inputs['number_of_monomers'])} entries. They must match."
+            )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)